catalog/core/catalog-core-impl/filter-proxy/src/main/java/ddf/catalog/filter/proxy/adapter/GeotoolsFilterAdapterImpl.java

```diff
@@ -134,6 +135,27 @@ public Object visit(Function expression, Object delegate) {
             throw new UnsupportedOperationException(
                     Function.class.getSimpleName() + " must not be null.");
         }
+
+        if (expression.getName().equals(ProximityFunction.NAME.getName())) {
+            List<Expression> parameters = expression.getParameters();
+            if (parameters.size() == ProximityFunction.NUM_PARAMETERS) {
+
+                String propertyName = parameters.remove(0)
```

Comment:
I'm not familiar with the `ExpressionVisitor` API - is it okay to remove parameters from the `Function`?